src/os/structs/nodeutils.js

```diff
@@ -0,0 +1,21 @@
+goog.provide('os.structs.nodeutils');
+
+
+/**
+ * Flattens a tree into a specified array - optionally filtering.
+ * @param {os.structs.ITreeNode} node
+ * @param {Array<os.structs.ITreeNode>} results
+ * @param {function(os.structs.ITreeNode): boolean} opt_filter
```

Comment:
The param name indicates this is optional but the type does not. Try `{function(os.structs.ITreeNode):boolean=}` to make it optional.

Reply:
Fixed